src/app/qgsmeasuredialog.cpp

```diff
@@ -43,7 +43,20 @@ QgsMeasureDialog::QgsMeasureDialog( QgsMeasureTool* tool, Qt::WindowFlags f )
   mMeasureArea = tool->measureArea();
   mTotal = 0.;
 
+  mUnitsCombo->addItem( tr( "Meters" ) );
+  mUnitsCombo->addItem( tr( "Feet" ) );
+  mUnitsCombo->addItem( tr( "Degrees" ) );
+  mUnitsCombo->addItem( tr( "Nautical Miles" ) );
+
+  QSettings settings;
+  QString units = settings.value( "/qgis/measure/displayunits", "meters" ).toString();
+  mUnitsCombo->setCurrentIndex( mUnitsCombo->findText( units, Qt::MatchFixedString ) );
+
   updateSettings();
+
+  connect( mUnitsCombo, SIGNAL( currentIndexChanged( const QString & ) ), this, SLOT( unitsChanged( const QString & ) ) );
+
+  mTable->headerItem()->setTextAlignment( 0, Qt::AlignRight );
```

Comment:
sure we can, by Qt Creator double clicking on header you can access to edit the tree widget properties and there you can set the text alignment.